### PR TITLE
Fix Null reference on SentryXamarinFormsIntegration.Register

### LINKED ISSUE #57: NullReferenceException in SentryXamarinFormsIntegration with slow Xamarin.Forms StartUp
### Environment

Which SDK and version?
Sentry 3.0.7
Sentry.Xamarin.Forms 1.0.2

### Steps to Reproduce

1. Starting Xamarin.Forms uwp app
2. Initalizing Xamarin.Forms takes a long time.
3. Runs into NullReferenceException (see screenshot)
4. App crashes

Forcing via Thread.Sleep for easy reproducing:

```
		protected override void OnLaunched(LaunchActivatedEventArgs e)
		{
			SentryXamarin.Init(options =>
			{
				options.Dsn = SENTRY_DSN;
				options.Release = Version.Major + "." + Version.Minor;
				options.SendDefaultPii = true;
				options.AddXamarinFormsIntegration();

			});

			System.Threading.Thread.Sleep(3000);
```

### Expected Result

No Exception should be raised

### Actual Result

See Exception screenshot:
error

Register() Method tries to call DiagnosticLogger, which is null, which causes the exception.

application is null, because start takes longer than 3000 ms in SentryXamarinFormsIntegration.cs:

```
        private async Task<Application> GetCurrentApplication()
        {
            for (int i = 0; i < 10 && Application.Current is null; i++)
            {
                await Task.Delay(300).ConfigureAwait(false);
            }
            return Application.Current;
        }
```

### Possible Solutions:

1. Run App on a faster computer ;-)

2. Add a Timeout / loop setting for application start. maybe reduce delay and increase loop times.

3. Add null check for  call "options.DiagnosticLogger.Log()"

### Workaround
Add options to Init:

```
	public class SentryLogger : IDiagnosticLogger
	{
		public bool IsEnabled(SentryLevel level)
		{
			return true;
		}

		public void Log(SentryLevel logLevel, string message, Exception exception = null, params object[] args)
		{
			Debug.WriteLine(message, args);
		}
	}


...
			SentryXamarin.Init(options =>
			{
				options.DiagnosticLogger = new SentryLogger();
				options.Debug = true;


...
			});
```

### PR DESCRIPTION
**Pull request by lucas-zimerman**
* Increased timeout for hooking to the current application from 3.5 seconds to 7 seconds.

Close #57

### CODE REVIEW

File: Src/Sentry.Xamarin.Forms/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Runtime.CompilerServices;
-[assembly: InternalsVisibleTo("Sentry.Xamarin.Forms.Tests.UWP")]
+[assembly: InternalsVisibleTo("Sentry.Xamarin.Forms.UWP.Tests")]
+[assembly: InternalsVisibleTo("Sentry.Xamarin.Forms.Testing")]
 
 [assembly: CLSCompliant(true)]
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin.Forms/Properties/AssemblyInfo.cs:6]: image

The missing line doesn't bother you? 😆 

The reason it's a good thing is that it avoids the `\n` diff on the line above when you add a new line to the file.

```suggestion
[assembly: CLSCompliant(true)]

```

`.editorconfig` rule helps with that as editors will always add a new line.

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
CI is unhappy though

**Comment by lucas-zimerman** [comment]
I'm not sure why it's unhappy
`_logs.Add(new (logLevel, message, exception, args));`
I'm adding a new tuple and it works on Windows, gotta find out if there's another way of adding a new tuple to a list

File: Src/Sentry.Xamarin.Forms/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (lucas-zimerman) [reply Src/Sentry.Xamarin.Forms/Properties/AssemblyInfo.cs]: It doesn't bother me but I'll keep that in mind on the next PRs

**Pull request merged**